examples/jetstream/queue-push-consumer/go/main.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+func main() {
+	// Use the env variable if running in the container, otherwise use the default.
+	url := os.Getenv("NATS_URL")
+	if url == "" {
+		url = nats.DefaultURL
+	}
+
+	// Create an unauthenticated connection to NATS.
+	nc, _ := nats.Connect(url)
+	defer nc.Drain()
+
+	// Access the JetStreamContext for managing streams and consumers
+	// as well as for publishing and subscription convenience methods.
+	js, _ := nc.JetStream()
+
+	// Declare a simple [limits-based stream][1] and populate the stream
+	// with a few messages.
+	// [1]: /examples/jetstream/limits-stream/go/
+	streamName := "EVENTS"
+
+	js.AddStream(&nats.StreamConfig{
+		Name:     streamName,
+		Subjects: []string{"events.>"},
+	})
+
+	// ### Ephemeral
+	// Like the standard [push consumer][1], the `JetStreamContext` provides
+	// a simple way to create an ephemeral queue push consumer. The only
+	// additional argument is the "group name".
+	// [1]: /examples/jetstream/push-consumer/go/
+	fmt.Println("# Ephemeral")
+	sub1, _ := js.QueueSubscribeSync("events.>", "event-processor", nats.AckExplicit())
```

Comment:
There are actually no ephemeral queue subscriptions right now... what happens is that the queue name becomes the durable name as well automatically:
https://github.com/nats-io/nats.go/blob/main/js.go#L1394-L1400

Reply:
:man_facepalming: I even wrote this in my blog post on queue push subscribers.. and I forgot that was the case. So the `Unsubscribe` to delete behavior still remains, but it won't be auto-cleaned up on the server. Thanks for catching.

Reply:
Do you have any insight as to why ephemerals are not supported?